src/main/java/com/atlauncher/viewmodel/base/settings/IModsSettingsViewModel.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * ATLauncher - https://github.com/ATLauncher/ATLauncher
+ * Copyright (C) 2013-2022 ATLauncher
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.atlauncher.viewmodel.base.settings;
+
+import com.atlauncher.data.AddModRestriction;
+import com.atlauncher.data.InstanceExportFormat;
+import com.atlauncher.data.ModPlatform;
+import com.atlauncher.evnt.listener.SettingsListener;
+import com.atlauncher.gui.tabs.settings.ModsSettingsTab;
+
+import io.reactivex.rxjava3.core.Observable;
+
+/**
+ * @since 2022 / 06 / 15
+ * <p>
+ * Why are there functions here that are acronym?
+ * The settings name would be too long otherwise.
+ * </p>
+ * <p>
+ * EAMBD : enableAddedModsByDefault
+ * <p>
+ * DCMOC : dontCheckModsOnCurseForge
+ * <p>
+ * DCMOM : dontCheckModsOnModrinth
+ * <p>
+ * ACFABF : allowCurseForgeAlphaBetaFiles
+ * <p>
+ * View model for {@link ModsSettingsTab}
+ */
+public interface IModsSettingsViewModel extends SettingsListener {
+
+    void setDefaultModPlatform(ModPlatform modPlatform);
+
+    Observable<Integer> DefaultModPlatformChanged();
+
+    void setAddModRestrictions(AddModRestriction modRestrictions);
+
+    Observable<Integer> AddModRestrictionsChanged();
+
+    void setEAMBD(Boolean b);
```

Comment:
Not a big fan of the abbreviations, makes it much harder to tell what these are actually for. I'd prefer to just have long names as there should be no limits on how long the names are

Reply:
Alright, will swap out with long names